irohad/network/impl/block_loader_service.cpp

```diff
@@ -56,21 +56,45 @@ grpc::Status BlockLoaderService::retrieveBlock(
                         "Bad hash provided");
   }
 
-  // required block must be in the cache
+  // try to fetch block from the consensus cache
   auto block = consensus_result_cache_->get();
-  if (not block) {
-    log_->info("Requested to retrieve a block from an empty cache");
-    return grpc::Status(grpc::StatusCode::NOT_FOUND, "Cache is empty");
-  }
-  if (block->hash() != hash) {
+  if (block) {
+    if (block->hash() == hash) {
+      auto transport_block =
+          std::static_pointer_cast<shared_model::proto::Block>(block)
+              ->getTransport();
+      response->CopyFrom(transport_block);
```

Comment:
Two copies of block are made here:
1. auto transport_block assignment
2. response CopyFrom

Consider using `*response = getTransport()` instead